design-proposals/klip-4-produce-data.md

```diff
@@ -0,0 +1,201 @@
+the values for the fields will be verified against the schema. If the columns contain `ROWKEY`, the
+value will be verified against the key schema but will not populate the corresponding field in the 
+value. Populating columns with `ROWTIME` will not be supported.
+
+### Serialization Support
+
+JSON, AVRO and DELIMITED value formats will all be supported, the latter of which will not accept
+complex types. Since we know the schema of the inserts, we can initially load the values into a 
+connect data type. We already have the proper serializers to convert from connect to each of the
+supported serialization formats.
+
+## Test plan
+
+This functionality is expected to work both in Interactive as well as Headless modes, and therefore
+all tests described below should be done in both:
+
+* Test that values can be serialized into JSON/AVRO/DELIMITED
+* Test that schemas are properly verified
+* Test `NULL` and table tombstones
+* Test `CREATE STREAM/TABLE` with topics that don't exist
+* Test producing data in topics that already exist
+* Test producing data to topics that are not associated with a KSQL source
+
+## Documentation Updates
+
+* The `CREATE TABLE` and `CREATE STREAM` syntax references will have an updated table properties
+section that includes the following rows:
+> ```rst
+> +=========================+======================================================================================================+
+> | KAFKA_TOPIC (required)  | The name of the Kafka topic that backs this source. The topic must either already exist in Kafka, or |
+> |                         | PARTITIONS and REPLICAS must be specified to create the topic. Command will fail if the topic exists |
+> |                         | with different partition/replica counts.                                                             |
+> +-------------------------+------------------------------------------------------------------------------------------------------+
+> | PARTITIONS              | The number of partitions in the backing topic. This property must be set if creating a SOURCE without|
+> |                         | an existing topic.                                                                                   |
+> +-------------------------+------------------------------------------------------------------------------------------------------+
+> | REPLICAS                | The number of replicas in the backing topic. This property must be set if creating a SOURCE without  |
+> |                         | an existing topic.                                                                                   |
+> +-------------------------+------------------------------------------------------------------------------------------------------+
+> ```
+* A new section for `INSERT INTO ... VALUES` will be added to the `sytnax.rst`:
+> ```rst
+> **Synopsis**
+> 
+> .. code:: sql
+>
+>   INSERT INTO stream_name/table_name [( column_name [, ...] )]
```

Comment:
To clarify, such a statement can only be used to insert a single event into a topic at a time, right? As in, if a user wanted to insert multiple events into the same topic, they'd have to issue multiple such statements, yes?

Reply:
Correct